src/System.Private.CoreLib/shared/System/Text/UTF32Encoding.cs

```diff
@@ -42,27 +42,27 @@ public sealed class UTF32Encoding : Encoding
         private static readonly byte[] s_bigEndianPreamble = new byte[4] { 0x00, 0x00, 0xFE, 0xFF };
         private static readonly byte[] s_littleEndianPreamble = new byte[4] { 0xFF, 0xFE, 0x00, 0x00 };
 
-        private bool _emitUTF32ByteOrderMark = false;
-        private bool _isThrowException = false;
-        private bool _bigEndian = false;
+        private readonly bool _emitUTF32ByteOrderMark;
+        private readonly bool _isThrowException;
+        private readonly bool _bigEndian;
 
 
-        public UTF32Encoding() : this(false, true, false)
+        public UTF32Encoding() : this(false, true)
         {
         }
 
 
         public UTF32Encoding(bool bigEndian, bool byteOrderMark) :
-            this(bigEndian, byteOrderMark, false)
+            base(bigEndian ? 12001 : 12000)
         {
+            _bigEndian = bigEndian;
+            _emitUTF32ByteOrderMark = byteOrderMark;
         }
 
 
         public UTF32Encoding(bool bigEndian, bool byteOrderMark, bool throwOnInvalidCharacters) :
```

Comment:
I assume that you are doing this change to make the constructor that takes `throwOnInvalidCharacters` argument trimmed by the linker. Is it correct?

Reply:
Correct linker or AOT